coap-gateway/service/config.go

```diff
@@ -93,13 +95,65 @@ func (c *AuthorizationConfig) Validate() error {
 	return nil
 }
 
+type InjectedTLSConfig struct {
+	NonIdentityCertificateEnabled bool `yaml:"nonIdentityCertificateEnabled" json:"nonIdentityCertificateEnabled"`
+}
+
+type InjectedCOAPConfig struct {
+	TLSConfig InjectedTLSConfig `yaml:"tls" json:"tls"`
+}
+
+func (c *InjectedCOAPConfig) Validate() error {
+	return nil
+}
+
 type COAPConfig struct {
 	coapService.Config         `yaml:",inline" json:",inline"`
 	ExternalAddress            string              `yaml:"externalAddress" json:"externalAddress"`
 	Authorization              AuthorizationConfig `yaml:"authorization" json:"authorization"`
 	OwnerCacheExpiration       time.Duration       `yaml:"ownerCacheExpiration" json:"ownerCacheExpiration"`
 	SubscriptionBufferSize     int                 `yaml:"subscriptionBufferSize" json:"subscriptionBufferSize"`
 	RequireBatchObserveEnabled bool                `yaml:"requireBatchObserveEnabled" json:"requireBatchObserveEnabled"`
+	InjectedCOAPConfig         InjectedCOAPConfig  `yaml:"-" json:"-"`
+}
+
+type COAPConfigMarshalerUnmarshaler struct {
+	COAPConfig `yaml:",inline"`
+}
+
+func (c *COAPConfigMarshalerUnmarshaler) UnmarshalYAML(value *yaml.Node) error {
+	err := value.Decode(&c.COAPConfig)
+	if err != nil {
+		return err
+	}
+	return value.Decode(&c.InjectedCOAPConfig)
+}
+
+func (c COAPConfigMarshalerUnmarshaler) MarshalYAML() (interface{}, error) {
```

Comment:
What's the difference between adding `nonIdentityCertificateEnabled` this way vs. simply adding it to COAPConfig?

Reply:
We don't want to change generic `COAPConfig` so we need to inject property to yaml file to proper place.